modules/DGM/Powell.cpp

```diff
@@ -173,6 +173,16 @@ namespace DirectGraphicalModels
 		} // infinite loop
 	}
 
+    vec_float_t CPowell::getParams(float (*objectiveFunct)(vec_float_t)) {
```

Comment:
I would argue that `getParams()` is exactly a getter. It gets the users the value stored in the private variable called `gBest`. I suggest letting `getParams(float)` there for backwards compatibility, but I believe it has certain flows, both conceptually and practically. In general, I think one would need the objective function in order to implement a class that inherits from `ParamEstAlgorithm` (i.e. another parameter optimization method). That is because they will need to call this function on some private member variables (as I did for PSO). Powell method seems to be a rare case where someone could update the parameters based solely on the objective function outcome called on those parameters.

Alternatively, we could provide getters for those private member variables, but from an OOP point of view, I believe these implementation details should remain "hidden".